File: backend/src/utils/scrubber.ts
const secretPatterns = [
  /(api|secret|token|key)[=:]\s*["']?([A-Za-z0-9_\-]{16,})/gi,
  /ghp_[A-Za-z0-9]{30,}/g,
  /sk-[A-Za-z0-9]{20,}/g
];

export function scrubSecrets(input: string): string {
  let output = input;

  for (const pattern of secretPatterns) {
    output = output.replace(pattern, "[REDACTED_SECRET]");
  }

  return output;
}

File: backend_flask/scrubber.py
import re

SECRET_PATTERNS = [
  re.compile(r"(api|secret|token|key)[=:]\s*[\"']?([A-Za-z0-9_\\-]{16,})", re.IGNORECASE),
  re.compile(r"ghp_[A-Za-z0-9]{30,}"),
  re.compile(r"sk-[A-Za-z0-9]{20,}")
]


def scrub_secrets(text: str) -> str:
  scrubbed = text
  for pattern in SECRET_PATTERNS:
    scrubbed = pattern.sub("[REDACTED_SECRET]", scrubbed)
  return scrubbed
